### PR TITLE
Contribution guide

### LINKED ISSUE #51: Add a contribution guide
Write a CONTRIBUTION.md file to define how contributors are meant to add code to the project.

### PR DESCRIPTION
**Pull request by tsjohns9**
closes #51 
Markdown file outlining how to contribute to the project for internal and external contributors.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Please make sure the following is done before submitting a pull request:
+
+1. Fork the repository and create your branch from master.
+2. If you’ve fixed a bug or added code that should be tested, then add tests.
+3. Ensure the test suite passes by running `make test`.
+4. Format your code with `make format`.
+5. Lint your code with `make lint`.
+6. If you haven’t already, complete the CLA.
+7. Make sure your pull request describes the issue you are fixing, or the feature you are adding.
```
Comment (chrisccoy) [inline CONTRIBUTING.md:63]: I feel like we should we provide better guidance around the issue/PR relationship. For example, "Fixes #"

**Review by Atraya21 — COMMENTED** [review]
Should we mention something about the criteria for accepting PR in the contributing guidelines?

**Review by sagarkal — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (tsjohns9) [reply CONTRIBUTING.md:63]: That's a good point. I'm working on adding a section for Axway contributors. For an OS contribution, not sure if they should add that comment, or if we should do it as the maintainers.

**Comment by tsjohns9** [comment]
@Atraya21 What do you have in mind?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+1. Labels are added to the pull request. These can be the same labels that are found on the issue.
+2. Assign the pull request to the 'Axway Mulesoft Agents' project. This is the board for tracking work in progress.
+3. Assign the pull request to a Milestone, which will typically be the current sprint.
+4. Link the pull request back to your issue.
+
+All of these steps can be taken care of after opening your pull request.
+
+Reviewers will automatically be added to your pull request. Assign the pull request to one of the core maintainers when you are ready to merge your branch. You may merge the branch once it has been approved.
+
+## Accepting pull requests
```
Comment (tsjohns9) [inline CONTRIBUTING.md:108]: @Atraya21 were you looking for something like this?

**Review by Atraya21 — APPROVED** [review]

**Pull request merged**